ee/clickhouse/models/test/test_cohort.py

```diff
@@ -321,6 +368,146 @@ def test_cohortpeople_action_basic(self):
         )
         self.assertEqual(len(results), 2)
 
+    def test_cohortpeople_timestamp(self):
+        action = _create_action(team=self.team, name="$pageview")
+        p1 = Person.objects.create(
+            team_id=self.team.pk,
+            distinct_ids=["1"],
+            properties={"$some_prop": "something", "$another_prop": "something"},
+        )
+
+        _create_event(
+            event="$pageview",
+            team=self.team,
+            distinct_id="1",
+            properties={"attr": "some_val"},
+            timestamp=datetime(2020, 1, 9, 12, 0, 1),
+        )
+
+        p2 = Person.objects.create(
+            team_id=self.team.pk,
+            distinct_ids=["2"],
+            properties={"$some_prop": "something", "$another_prop": "something"},
+        )
+
+        _create_event(
+            event="$pageview",
+            team=self.team,
+            distinct_id="2",
+            properties={"attr": "some_val"},
+            timestamp=datetime(2020, 1, 7, 12, 0, 1),
+        )
+
+        cohort1 = Cohort.objects.create(
+            team=self.team,
+            groups=[{"action_id": action.pk, "start_date": datetime(2020, 1, 8, 12, 0, 1)}],
+            name="cohort1",
+        )
+        with freeze_time("2020-01-10"):
+            cohort1.calculate_people_ch()
+
+        results = sync_execute("SELECT person_id FROM cohortpeople")
+        self.assertEqual(len(results), 1)
+
+    def test_cohortpeople_action_count(self):
```

Comment:
Extract the mock code into a method with a good name.

Reply:
Overall, I feel this method could be broken up into smaller chunks.

Reply:
done